README_local_MacOS.md

```diff
@@ -0,0 +1,122 @@
+# Example Local Dev Environment Setup
+
+## MacOS
+
+* `Go` version from
+  [go.mod](https://github.com/gravitational/teleport/blob/master/go.mod#L3)
+  
+  * Follow [official instructions](https://go.dev/doc/install) to install `Go`
+    * **On an M1 Mac, download ARM64 installer from https://go.dev/dl/**
+    * Download the installer for `<version from go.mod>`  
+    * After installing, don't forget to `export PATH="/usr/local/go/bin:$PATH"` in `~/.zprofile`
+    * If you need other go versions, see https://go.dev/doc/manage-install
+      * You will need to add `export PATH="$HOME/go/bin:$PATH"` to the `~/.zprofile`
+
+  * Or install required version of `Go` with homebrew:
+
+  ```shell
+  # if we are not on the latest, you might need to
+  # $ brew install go@<version from go.mod>, i.e. 1.16
+  #
+  # check which version will be installed by running
+  # $ brew info go
+  
+  $ brew install go
+  ````
+
+* `Rust` and `Cargo` version from
+  [build.assets/Makefile](https://github.com/gravitational/teleport/blob/master/build.assets/Makefile#L21)
+  (search for RUST_VERSION):
+
+  * Follow [official instructions](https://www.rust-lang.org/tools/install) to install `rustup`
+  *  Or install with homebrew:
+  
+  ```shell
+  $ brew install rustup
+  ```
+  
+  * Initialize Rustup
+  
+  ```shell
+  $ rustup-init
+  # accept defaults
+  #
+  # Once command finishes successfully, you might need to add
+  # 
+  # export PATH="$HOME/.cargo/bin:$PATH"
+  # 
+  # into ~/.zprofile and run
+  # 
+  # $ . ~/.zprofile
+  # 
+  # or open a new shell
+  ```
+  
+  * If on M1 Mac, add `x86_64` target 
```

Comment:
Why?

Reply:
https://github.com/gravitational/teleport/blob/master/Makefile#L136

Reply:
`make rdpclient` will build the RDP client for the host platform, so it will do the right thing on ARM or Intel macs.

You should not need to add the x86_64 target.

Reply:
```
╭─ ~/Documents/code/Teleport/teleport  dima/readme-update-libfido-build ⇣17 +1 !2 ▓▒░·········································································░▒▓ ✔  4s  16:44:15 ─╮
╰─ make test-go                                                                                                                                                                     ─╯
cargo build -p rdp-client --release --target=x86_64-apple-darwin
   Compiling cfg-if v1.0.0
   Compiling byteorder v1.4.3
   Compiling spin v0.5.2
   Compiling ppv-lite86 v0.2.16
   Compiling base64ct v1.5.0
   Compiling arrayvec v0.5.2
   Compiling ryu v1.0.10
   Compiling static_assertions v1.1.0
   Compiling subtle v2.4.1
   Compiling bitflags v1.3.2
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-apple-darwin` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-apple-darwin`

For more information about this error, try `rustc --explain E0463`.
error: could not compile `cfg-if` due to previous error
warning: build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-apple-darwin` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-apple-darwin`

error: build failed
make: *** [rdpclient] Error 101
```

I am thinking this had something to do with the fact that `Go` was installed for `amd64` architecture. Now that I run `make test-go`, it indeed compiles everything correctly. I cleaned this up. Thanks!